client/lib/site-specific-plans-details-list/assembler.js

```diff
@@ -7,7 +7,7 @@ function createSiteSpecificPlanObject( plan ) {
 	return {
 		currentPlan: Boolean( plan.current_plan ),
 		expiry: plan.expiry,
-		expiryMoment: moment( plan.expiry ),
+		expiryMoment: moment( plan.expiry ).startOf( 'day' ),
```

Comment:
@Tug The issue we encountered during the meetup is simplified a little bit by:

- Ensuring that **all** moments (those in this assembler and the current moment calculated with `moment()`) are set to the start of the day.
- Using `moment().startOf( 'day' )` instead of `moment().set( { hour: 0, ... } )` each time.